api/src/data/resolvers/mutations/widgets.ts

```diff
@@ -422,51 +428,71 @@ const widgetMutations = {
       companyData,
       data,
       cachedCustomerId,
-      deviceToken
+      deviceToken,
+      visitorId
     } = args;
 
     const customData = data;
 
     // find brand
-    const brand = await getBrand(brandCode);
-
-    if (!brand) {
-      throw new Error('Brand not found');
-    }
+    const brand = await Brands.getBrand({ code: brandCode });
 
     // find integration
     const integration = await getIntegration({
       brandId: brand._id,
-      type: 'messenger',
+      type: KIND_CHOICES.MESSENGER,
       callback: async () => {
-        return Integrations.getWidgetIntegration(brandCode, 'messenger');
+        return Integrations.getWidgetIntegration(
+          brandCode,
+          KIND_CHOICES.MESSENGER
+        );
       }
     });
 
-    let customer = await Customers.getWidgetCustomer({
-      integrationId: integration._id,
-      cachedCustomerId,
-      email,
-      phone,
-      code
-    });
+    let customer;
 
-    const doc = {
-      integrationId: integration._id,
-      email,
-      phone,
-      code,
-      isUser,
-      deviceToken
-    };
+    if (cachedCustomerId || email || phone || code) {
+      customer = await Customers.getWidgetCustomer({
+        integrationId: integration._id,
+        cachedCustomerId,
+        email,
+        phone,
+        code
+      });
 
-    customer = customer
-      ? await Customers.updateMessengerCustomer({
-          _id: customer._id,
-          doc,
+      const doc = {
+        integrationId: integration._id,
+        email,
+        phone,
+        code,
+        isUser,
+        deviceToken
+      };
+
+      customer = customer
+        ? await Customers.updateMessengerCustomer({
+            _id: customer._id,
+            doc,
+            customData
+          })
+        : await Customers.createMessengerCustomer({ doc, customData });
+    }
+
+    if (visitorId) {
+      const isRunning = await isLoggerRunning();
+
+      if (isRunning) {
+        await sendToVisitorLog(
```

Comment:
why can not we catch the exception in sendToVisitorLog without calling isLoggerRunning. ?